api/models/joi-schemas.ts

```diff
@@ -37,3 +35,22 @@ export const updateLinkSchema = Joi.object({
   ),
   enabled: Joi.bool(),
 }).xor("longUrl", "enabled");
+
+export const createLinkSchemaApp = createLinkSchema.keys({
+  longUrl: Joi.string()
+    .regex(
+      /^(?:(http|https|ftp)?:\/\/)?[\w.-]+(?:\.[\w\.-]+)+[\w\-\._~:/?#[\]%@!\$&'\(\)\*\+,;=.]+$/
+    )
+    .required()
+    .trim(),
+  customCode: Joi.string()
+    .regex(/^(?=.*[A-Za-z])[^-_][-\w]{3,25}$/)
+    .trim(),
+  linkIds: Joi.array().items(Joi.string().regex(/^[A-Za-z]{12}$/)),
```

Comment:
why is `linkIds` required to create a link?, `linkId` will be generated by the api no?

Reply:
we don't, we append the created linkId into the `linkIds` array if present and return 

we do the same in web routes, just we get the linkIds from the cookies

Reply:
ahh, so the `linkIds` are not stored locally? but instead sent when a new link is created, it is appended with the old `linkIds` and sent back which is the new state?